reconcile/glitchtip_project_alerts/integration.py

```diff
@@ -69,35 +69,44 @@ def fetch_desired_state(
                 glitchtip_project.organization.name,
                 Organization(name=glitchtip_project.organization.name),
             )
+            alerts = []
+            for alert in glitchtip_project.alerts or []:
+                project_alert_recipients: list[ProjectAlertRecipient] = []
+                for recp in alert.recipients:
+                    if isinstance(recp, GlitchtipProjectAlertRecipientEmailV1):
+                        project_alert_recipients.append(
+                            ProjectAlertRecipient(recipient_type=RecipientType.EMAIL)
+                        )
+                    elif isinstance(recp, GlitchtipProjectAlertRecipientWebhookV1):
+                        url = recp.url
+                        if not url and recp.url_secret:
+                            url = self.secret_reader.read_secret(recp.url_secret)
+                        if not url:
+                            raise ValueError(
+                                "url or urlSecret must be set for webhook recipient"
+                            )
+                        project_alert_recipients.append(
+                            ProjectAlertRecipient(
+                                recipient_type=RecipientType.WEBHOOK,
+                                url=url,
+                            )
+                        )
```

Comment:
Since this become a little complex now, how about extract a method for it?

```python
    def build_project_alert_recipient(self, recipient):
        match recipient:
            case GlitchtipProjectAlertRecipientEmailV1():
                return ProjectAlertRecipient(recipient_type=RecipientType.EMAIL)
            case GlitchtipProjectAlertRecipientWebhookV1(
                url=url,
                url_secret=url_secret,
            ):
                if not url and url_secret:
                    url = self.secret_reader.read_secret(url_secret)
                if not url:
                    raise ValueError(
                        "url or urlSecret must be set for webhook recipient"
                    )
                return ProjectAlertRecipient(
                    recipient_type=RecipientType.WEBHOOK,
                    url=url,
                )
```

Then it can be used similar as before:

```python
            alerts = [
                ProjectAlert(
                    name=alert.name,
                    timespan_minutes=alert.timespan_minutes,
                    quantity=alert.quantity,
                    recipients=list(
                        map(self.build_project_alert_recipient, alert.recipients)
                    ),
                )
                for alert in glitchtip_project.alerts or []
            ]
```

Reply:
Thanks for your suggestion. I applied it with [f5c5559](https://github.com/app-sre/qontract-reconcile/pull/3786/commits/f5c5559af4c0c45173ee253605aa7d90c4d479d7)